adapters/eplanning/eplanning.go

```diff
@@ -61,12 +66,47 @@ type hbResponseAd struct {
 	Height       uint64 `json:"h,omitempty"`
 }
 
+type byPriority []openrtb.Format
+
+func (a byPriority) Len() int      { return len(a) }
+func (a byPriority) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byPriority) Less(i, j int) bool {
+	var priorityOrderForSizesAsc []string
+	if isMobile > 0 {
+		priorityOrderForSizesAsc = priorityOrderForMobileSizesAsc
+	} else {
+		priorityOrderForSizesAsc = priorityOrderForDesktopSizesAsc
+	}
+	size1 := fmt.Sprintf("%dx%d", a[i].W, a[i].H)
+	size2 := fmt.Sprintf("%dx%d", a[j].W, a[j].H)
+	index1 := indexOf(size1, priorityOrderForSizesAsc)
+	index2 := indexOf(size2, priorityOrderForSizesAsc)
+	if index1 > -1 {
+		if index2 > -1 {
+			if index1 < index2 {
+				return false
+			} else {
+				return true
+			}
+		} else {
+			return false
+		}
+	} else {
+		if index2 > -1 {
+			return true
+		} else {
+			return false
+		}
+	}
+}
```

Comment:
Sorting algorithms are always fun to read. Could you please describe the algorithm to assist with the review?

I'm a bit concerned with the performance of building the `size1` and `size2` strings on each compare. Perhaps that burden can be reduced with memoization.. although before I perhaps prematurely worry about performance, how many formats do you typically see in requests?